recipe-box-backend/src/main/java/com/endava/recipebox/service/impl/RecipeServiceImpl.java

```diff
@@ -98,4 +99,36 @@ public String createRecipe(RecipeAddRequestDTO recipeAddRequestDTO, Long userId)
 
         return "Recipe added successfully";
     }
+
+    @Override
+    @Transactional
+    public String editRecipe(RecipeEditRequestDTO recipeAddRequestDTO, Long userId){
+        Recipe recipe = recipeRepository.findById(recipeAddRequestDTO.getId())
+                .orElseThrow(() -> new EntityNotFoundException("Recipe not found"));
+
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("User not found"));
+
+        boolean isAdmin = user.getRole().equals(Role.Admin);
+        boolean isOwner = recipe.getUser().getId().equals(userId);
+
+        if (!isAdmin && !isOwner) {
```

Comment:
so only when user is admin and owner is allowed to edit the recipe ? What is happening when user is admin but not owner of recipe?